assets/src/edit-story/components/richText/karma/inlineSelection.js

```diff
@@ -0,0 +1,366 @@
+        await data.fixture.events.click(bold.button);
+
+        // Verify bold is now on and font weight is Bold
+        expect(bold.checked).toBe(true);
+        expect(fontWeight.value).toBe('Bold');
+
+        // Exit edit-mode
+        await data.fixture.events.keyboard.press('Escape');
+
+        // Assume text content to be correctly formatted
+        const actual = getTextContent();
+        const expected = `<span style="font-weight: 900">F</span><span style="font-weight: 700">il</span>l in some text`;
+        expect(actual).toBe(expected);
+      });
+
+      it('should make black+bold+regular selection black when toggling', async () => {
+        const {
+          bold,
+          fontWeight,
+        } = data.fixture.editor.inspector.designPanel.textStyle;
+
+        // Select first three characters (900, 700 and 400)
+        await setSelection(0, 3);
+
+        // Check that bold toggle is off but font weight is "multiple"
+        expect(bold.checked).toBe(false);
+        expect(fontWeight.value).toBe('(multiple)');
+
+        // Toggle it by pressing the bold button
+        await data.fixture.events.click(bold.button);
+
+        // Verify bold is now on and font weight is Black
+        expect(bold.checked).toBe(true);
+        expect(fontWeight.value).toBe('Black');
+
+        // Exit edit-mode
+        await data.fixture.events.keyboard.press('Escape');
+
+        // Assume text content to be correctly formatted
+        const actual = getTextContent();
+        const expected = `<span style="font-weight: 900">Fil</span>l in some text`;
+        expect(actual).toBe(expected);
+      });
+    });
+
+    it('should apply global formats (here line height) even when a selection is present', async () => {
+      // Expect line height to be default "Roboto"
+      const getDisplayTextStyles = () => {
+        const displayNode = data.fixture.editor.canvas.displayLayer.display(
+          data.textId
+        ).node;
+        const paragraph = displayNode.querySelector('p');
+        return window.getComputedStyle(paragraph);
+      };
+      const initialLineHeight = parseFloat(getDisplayTextStyles().lineHeight);
+
+      const {
+        lineHeight,
+      } = data.fixture.editor.inspector.designPanel.textStyle;
+
+      // Enter edit-mode
+      await data.fixture.events.keyboard.press('Enter');
+
+      // Select something
+      await setSelection(5, 7);
+
+      // Change line height to 5
+      await data.fixture.events.click(lineHeight, { clickCount: 3 });
+      await data.fixture.events.keyboard.type('5');
+      await data.fixture.events.keyboard.press('Escape');
+
+      // Exit edit-mode
+      await data.fixture.events.keyboard.press('Escape');
+
+      // Expect text content to be unchanged
+      expect(getTextContent()).toBe('Fill in some text');
+
+      // Expect line height to have changed
+      expect(getDisplayTextStyles().lineHeight).not.toBe(initialLineHeight);
```

Comment:
Any reason for not checking the specific chosen line-height value?

Reply:
Yes, it's some weird value not straight-forward related to the actual input (the computed style is in pixels where the input value is a relative number). Also, this is not a test of line height actually working correctly, just that the text field correctly picks up changes to non-inline styles even while in edit mode.